src/csharp/Extensions/Microsoft.Spark.Extensions.DotNet.Interactive/PackageHelper.cs

```diff
@@ -0,0 +1,143 @@
+                {
+                    // probePath.FullName
+                    //   /path/to/packages/package.name/package.version/
+                    // resolvedPackage.PackageRoot
+                    //   /path/to/packages/package.name/package.version/
+                    // GetRelativeToPackages(..)
+                    //   package.name/package.version
+                    nativeProbingPaths.Add(
+                        GetPathRelativeToPackages(
+                            probePath.FullName,
+                            resolvedPackage.PackageRoot));
+                }
+
+                nugetMetadata.Add(
+                    new DependencyProviderUtils.NuGetMetadata
+                    {
+                        FileName = package.NuGetFile.Name,
+                        PackageName = resolvedPackage.PackageName,
+                        PackageVersion = resolvedPackage.PackageVersion
+                    });
+
+                yield return package.NuGetFile.FullName;
+            }
+
+            if (nugetMetadata.Count > 0)
+            {
+                var metadataPath =
+                    Path.Combine(
+                        writePath,
+                        DependencyProviderUtils.CreateFileName(
+                            Interlocked.Increment(ref _metadataCounter)));
+                new DependencyProviderUtils.Metadata
+                {
+                    AssemblyProbingPaths = assemblyProbingPaths.ToArray(),
+                    NativeProbingPaths = nativeProbingPaths.ToArray(),
+                    NuGets = nugetMetadata.ToArray()
+                }.Serialize(metadataPath);
+
+                yield return metadataPath;
+            }
+        }
+
+        /// <summary>
+        /// Return the delta of the list of packages that have been introduced
+        /// since the last call.
+        /// </summary>
+        /// <returns>The delta of the list of packages.</returns>
+        private IEnumerable<ResolvedNuGetPackage> GetNewPackages()
+        {
+            IEnumerable<ResolvedPackageReference> packages =
+                _packageRestoreContextWrapper.ResolvedPackageReferences;
+            foreach (ResolvedPackageReference package in packages)
+            {
+                IEnumerable<FileInfo> files =
+                    package.PackageRoot.EnumerateFiles("*.nupkg", SearchOption.AllDirectories);
+
+                foreach (FileInfo file in files)
+                {
+                    if (_filesCopied.TryAdd(file.Name, 1))
+                    {
+                        yield return new ResolvedNuGetPackage
+                        {
+                            ResolvedPackage = package,
+                            NuGetFile = file
+                        };
+                    }
+                }
+            }
+        }
+
+        private string GetPathRelativeToPackages(string file, DirectoryInfo directory)
+        {
+            string strippedRoot = file
```

Comment:
Can you add one example for this function please?

Reply:
Should I move the examples that are at the callsite of GetPathRelativeToPackages(..) and add it to the method description instead ?

Reply:
oh, you can just add it here.

Reply:
just relevant example related to this function should be fine.